internal/report/report.go

```diff
@@ -433,7 +433,19 @@ func PrintAssembly(w io.Writer, rpt *Report, obj plugin.ObjTool, maxFuncs int) e
 	}
 
 	if len(syms) == 0 {
-		return fmt.Errorf("no matches found for regexp: %s", o.Symbol)
+		if address == nil { // User provides a symbol instead of an address
+			if len(symbols) == 0 {
+				return fmt.Errorf("no matches found for regexp %s in binaries", o.Symbol)
+			} else {
+				return fmt.Errorf("regex %s found in binaries, but not in the profile", o.Symbol)
```

Comment:
I wonder if it would make sense to simplify this message. It tries to be specific but I think it is may be too specific. Just saying something like "no data matched regexp %s" or something like that might be enough.

Reply:
The goal of this PR is to tell that user about the reason why there is no match: the provided symbol/address did not match the binary or did not match anything in the profile. I feel like simplifying the message would defeat the purpose